lib/active_model/serializer/adapter/json.rb

```diff
@@ -7,37 +7,13 @@ class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
           if serializer.respond_to?(:each)
-            @result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
```

Comment:
Also, `result` is very generic. `hash` might not be the best, but at least it's slightly more precise (and given the method name is `serializable_hash` it does make sense).